src/analyses/dirty.h

```diff
@@ -17,42 +17,74 @@ Date: March 2013
 #include <unordered_set>
 
 #include <util/std_expr.h>
+#include <util/invariant.h>
 #include <goto-programs/goto_functions.h>
 
 class dirtyt
 {
+private:
+  void die_if_uninitialized() const
+  {
+    INVARIANT(initialized,
+        "Uninitialized dirtyt. This dirtyt was constructed using the default "
+        "constructor and not subsequently initialized using "
+        "dirtyt::build().");
+  }
+
 public:
+  bool initialized;
   typedef std::unordered_set<irep_idt, irep_id_hash> id_sett;
   typedef goto_functionst::goto_functiont goto_functiont;
 
-  explicit dirtyt(const goto_functiont &goto_function)
+  dirtyt():
+    initialized(false)
+  {
```

Comment:
It's no exactly clear why this is needed.  I guess there must be some reason why you have to create and initialise it separately.